src/PiWeb.Import.Sdk/Modules/ImportFormat/FileSourceExtensions.cs

```diff
@@ -57,20 +57,43 @@ public static bool HasAnyExtension(this IFileSource fileSource, IEnumerable<stri
     /// <param name="fileSources">The <see cref="IFileSource"/>s to check.</param>
     /// <param name="extensions">The file extensions (including the period ".").</param>
     /// <returns>The first matching <see cref="IFileSource"/>; otherwise <c>null</c>.</returns>
-    public static IFileSource? GetFirstWithExtension(this IEnumerable<IFileSource> fileSources, ISet<string> extensions)
+    public static IFileSource? GetFirstWithAnyExtension(this IEnumerable<IFileSource> fileSources, IEnumerable<string> extensions)
     {
         return fileSources.FirstOrDefault(fileSource => fileSource.HasAnyExtension(extensions));
     }
 
     /// <summary>
-    /// Gets the first <see cref="IFileSource"/> within <paramref name="fileSources"/> that has the requested <paramref name="extensions"/>.
+    /// Gets the first <see cref="IFileSource"/> within <paramref name="fileSources"/> that has the requested <paramref name="extension"/>.
     /// </summary>
     /// <param name="fileSources">The <see cref="IFileSource"/>s to check.</param>
-    /// <param name="extensions">The file extension (including the period ".").</param>
+    /// <param name="extension">The file extension (including the period ".").</param>
     /// <returns>The first matching <see cref="IFileSource"/>; otherwise <c>null</c>.</returns>
-    public static IFileSource? GetFirstWithExtension(this IEnumerable<IFileSource> fileSources, string extensions)
+    public static IFileSource? GetFirstWithExtension(this IEnumerable<IFileSource> fileSources, string extension)
+    {
+        return fileSources.FirstOrDefault(fileSource => fileSource.HasExtension(extension));
+    }
+
+    /// <summary>
+    /// Gets all <see cref="IFileSource"/>s within <paramref name="fileSources"/> that have any of the requested file extension
+    /// in <paramref name="extensions"/>.
+    /// </summary>
+    /// <param name="fileSources">The <see cref="IFileSource"/>s to check.</param>
+    /// <param name="extensions">The file extensions (including the period ".").</param>
+    /// <returns>The list of <see cref="IFileSource"/>s with the <paramref name="extensions"/>.</returns>
+    public static IEnumerable<IFileSource> AllWithExtensions(this IEnumerable<IFileSource> fileSources, IEnumerable<string> extensions)
```

Comment:
I would say we should return `IReadOnlyList` here, which is just more comfortable to use and we also already called `ToList()` anyway.

Reply:
It might be nice to be able to generate this lazily though. On the other hand, I think we have to refactor this again anyway to make it work better for the planner. So for PR this is probably not too important.

Reply:
I changed it to ReadOnlyList